src/tcgwars/logic/impl/gen4/PopSeries9.groovy

```diff
@@ -199,12 +215,14 @@ public enum PopSeries9 implements LogicCardInfo {
           move "Poltergeist", {
             text "30+ damage. Look at your opponent’s hand. This attack does 30 damage plus 10 more damage for each Trainer, Supporter, and Stadium card in your opponent’s hand."
             energyCost P, C
-            attackRequirement {}
+            attackRequirement {
+              assert opp.hand : "Opponent's hand is empty"
+            }
             onAttack {
-              damage 0
+              def filteredHand = opp.hand.shuffledCopy().showToMe("Opponent's hand").filterByType(ITEM, SUPPORTER, STADIUM)
```

Comment:
Hidden selections are shuffled, though since those are the only ones that it matters for, yeah shuffling the copy you see should no longer be necessary.